File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/digitaltwin/GetLinkResponse.java
package org.opcfoundation.webserver.types.message.digitaltwin;

import org.opcfoundation.webserver.types.DigitalTwinDescriptor;
import org.opcfoundation.webserver.types.ReferenceTargetDescriptor;

import java.util.ArrayList;
import java.util.List;

public class GetLinkResponse {
    private final List<ReferenceTargetDescriptor> targets;
    private boolean containsMoreData;

    public GetLinkResponse()
    {
        this.targets = new ArrayList<>();
        containsMoreData = false;
    }

    public void add(ReferenceTargetDescriptor descriptor)
    {
        targets.add(descriptor);
    }

    public List<ReferenceTargetDescriptor> getTargets() {
        return targets;
    }

    public boolean containsMoreData()
    {
        return containsMoreData;
    }

    public void setContainsMoreData(boolean containsMoreData) {
        this.containsMoreData = containsMoreData;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/GetLinkResponse.ts
import { ReferenceTargetDescriptor } from "../../../types/digitaltwin/ReferenceTargetDescriptor";

export class GetLinkResponse {
    private readonly _targets: Array<ReferenceTargetDescriptor>;
    private _containsMoreData: boolean;

    constructor() {
        this._targets = [];
        this._containsMoreData = false;
    }

    add(descriptor: ReferenceTargetDescriptor): void {
        this._targets.push(descriptor);
    }

    get targets(): Array<ReferenceTargetDescriptor> {
        return this._targets;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }
}